docs/content/v0.20/architecture.md

```diff
@@ -1,36 +1,479 @@
 # KubeStellar Architecture
 
-**NOTE**: This is only a very rudimentary start.
+KubeStellar provides multi-cluster deployment of Kubernetes objects, controlled by a simple 
+placement policy, where Kubernetes objects are expressed in their native format with no wrapping
+or bundling. The high-level architecture for KubeStellar is illustrated in Figure 1.
 
-## Higher Level
+<figure>
+  <img src="assets/high-level-architecture.png"  alt="High Level Architecture">
+  <figcaption align="center">Figure 1 - High Level Architecture </figcaption>
+</figure>
 
-At the higher level there are three modules.
+KubeStellar relies on the concept of *spaces*, which are abstractions 
+that represent a Kubernetes API Server and its controllers. Users can use 
+spaces to perform these tasks:
 
-1. KubeStellar
-2. [the status addon](https://github.ibm.com/dettori/status-addon)
-3. KubeFlex
+1. Create *Workload Definition Spaces* (WDSs) to store the definitions of their workloads.
+2. Create *Inventory and Transport Spaces* (ITSs) to manage the inventory of clusters and 
+the transport of workloads.
+3. Register and label Workload Execution Clusters (WECs) with the Inventory and 
+Transport Space, to keep track of the available clusters and their characteristics.
+4. Define Placement Policies to specify *what" workloads and *where* should be 
+deployed on the WECs.
+5. Submit Workload Definitions in the native Kubernetes format to the WDSs, 
+and let the Placement Policies govern which WECs should receive them.
+6. Check the status of submitted workloads from the WDS.
 
-The last two are independent and KubeStellar depends on them.
+KubeStellar users can have different roles and responsibilities. 
```

Comment:
If we want to define roles, we should have a special subsection for roles with description of the different operations each role can perform and how the roles can be assigned

Reply:
@MikeSpreitzer suggested not to list all the ops for roles for now, as we may get stuck into that discussion for a while, so I simplified that section to just mention that there are different roles and when we talk about a "user" he may have all of them.